src/main/java/com/segment/analytics/android/integrations/firebase/FirebaseIntegration.java

```diff
@@ -3,88 +3,242 @@
  */
 public class FirebaseIntegration extends Integration<FirebaseAnalytics> {
 
-  public static FirebaseAnalytics mFirebaseAnalytics;
-
-  public static final Factory FACTORY = new Factory() {
-    @Override public Integration<?> create(ValueMap settings, Analytics analytics) {
-      Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
-        logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
-        return null;
-      }
-      if (!hasPermission(analytics.getApplication(), Manifest.permission.INTERNET)) {
-        logger.debug("INTERNET is required for Firebase Analytics.");
-        return null;
-      }
+  public static final Factory FACTORY =
+      new Factory() {
+        @Override
+        public Integration<?> create(ValueMap settings, Analytics analytics) {
+          Logger logger = analytics.logger(FIREBASE_ANALYTICS_KEY);
+          if (!hasPermission(
+              analytics.getApplication(), Manifest.permission.ACCESS_NETWORK_STATE)) {
+            logger.debug("ACCESS_NETWORK_STATE is required for Firebase Analytics.");
+            return null;
+          }
+          if (!hasPermission(analytics.getApplication(), Manifest.permission.WAKE_LOCK)) {
+            logger.debug("WAKE_LOCK is required for Firebase Analytics.");
+            return null;
+          }
+
+          Context context = analytics.getApplication();
+
+          return new FirebaseIntegration(context, logger);
+        }
+
+        @Override
+        public String key() {
+          return FIREBASE_ANALYTICS_KEY;
+        }
+      };
+
+  private static final String FIREBASE_ANALYTICS_KEY = "Firebase";
+  final Logger logger;
+  final FirebaseAnalytics firebaseAnalytics;
+  private static final Map<String, String> eventMapper = createEventMap();
+
+  private static Map<String, String> createEventMap() {
+    Map<String, String> eventMapper = new HashMap<>();
+    eventMapper.put("Product Added", Event.ADD_TO_CART);
+    eventMapper.put("Checkout Started", Event.BEGIN_CHECKOUT);
+    eventMapper.put("Order Completed", Event.ECOMMERCE_PURCHASE);
+    eventMapper.put("Order Refunded", Event.PURCHASE_REFUND);
+    eventMapper.put("Product Viewed", Event.VIEW_ITEM);
+    eventMapper.put("Product List Viewed", Event.VIEW_ITEM_LIST);
+    eventMapper.put("Payment Info Entered", Event.ADD_PAYMENT_INFO);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+    eventMapper.put("Product Added to Wishlist", Event.ADD_TO_WISHLIST);
+    eventMapper.put("Product Shared", Event.SHARE);
+    eventMapper.put("Product Clicked", Event.SELECT_CONTENT);
+    eventMapper.put("Product Searched", Event.SEARCH);
+    eventMapper.put("Promotion Viewed", Event.PRESENT_OFFER);
+    return eventMapper;
+  }
 
-      Context context = analytics.getApplication();
+  private static final Map<String, String> propertyMapper = createPropertyMap();
```

Comment:
Constants should be constant cased https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names.